### ISSUE TITLE
XGBoosterLoadModel() causes subsequent UpdateOneIter() to hang

### ISSUE THREAD

**Issue by joetigger** [dmlc/xgboost#1889]
I am using xgboost (#da2556f58a122efcf7034d955e21040843a3b8a) on 64-bit centOS7. It compiles and installs fine, but I can't get the CLI to work properly. Here's the snippet of my code:

```
for (unsigned epoch=0; epoch<100; epoch++) {
  XGBoosterUpdateOneIter(ensemble, epoch, matrix);
  XGBoosterEvalOneIter(ensemble, epoch, validate, validate_name, validate_size, &result);
  if (validate_size > 1) {
    const char* p = strchr(result, ':');
    if (!p) continue;
    p = strchr(p+1, ':');
    if (!p) continue;
    double rmse = atof(p+1);
    if (rmse >= prev_rmse) {
      XGBoosterSetParam(ensemble, "eta", to_string(reduce(eta)).c_str());
      assert(!XGBoosterLoadModel(ensemble, model));
    } else {
      assert(!XGBoosterSaveModel(ensemble, model));
    }
    prev_rmse = rmse;
  }
}
````
Basically I'm trying to use diminishing learning rate as I add more trees, but the code always hang after I adjust eta 1-3 times. I also tried XGBoosterGetModelRaw() and XGBoosterLoadModelFromBuffer() and it hang the same way. However, if I comment out XGBoosterLoadModel() then it's OK.

How am I supposed to use XGBoosterLoadModel()?

**Comment by pommedeterresautee** [comment]
Old question, no answer. Closing
